sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/SplitUpdateAsDeleteAndInsert.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.plans.logical
+
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeSet, Expression}
+import org.apache.spark.sql.catalyst.util.truncatedString
+
+case class SplitUpdateAsDeleteAndInsert(
```

Comment:
Maybe, I can use `Expand`/`ExpandExec`. Let me explore this to see if there is any performance implication.

Reply:
Okay, I do think this node is a bit more efficient compared to `Expand` as it acts almost like `Project`. 

- No need to call `copy()` on each row.
- Sub expression elimination like in `ProjectExec`.
- Deferred evaluation of input attributes unless they are needed like in `ProjectExec`.

What are your thoughts, do you think it is worth having such a node? Technically, `Expand` would work too but it will be less efficient. 

cc @dongjoon-hyun @viirya @cloud-fan @sunchao @huaxingao

Reply:
Maybe, we can implement some of those optimizations for Expand instead of adding another node. Anyway, let me know.

Reply:
Switched to `Expand`, will follow up with improvements later.